src/Data/HashMap/Linear.hs

```diff
@@ -231,15 +231,15 @@ probeFrom (k, p) ix (HashMap sz@(Size s) ct arr) = read arr ix & \case
 -- # Querying
 --------------------------------------------------
 
-size :: HashMap k v #-> (HashMap k v, Int)
-size (HashMap sz ct@(Count c) arr) = (HashMap sz ct arr, c)
+size :: HashMap k v #-> (HashMap k v, Unrestricted Int)
```

Comment:
Technically should be in a different PR but don't bother right now. I think it's cumbersome so I'll approve this one but going forward that's a good idea and cleaner.